File: Classes/EVEOnlineAPI/Character/EVEMailMessages.m
//
//  EVEMailMessages.m
//  EVEOnlineAPI
//
//  Created by Artem Shimanski on 6/4/10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "EVEMailMessages.h"


@implementation EVEMailMessagesItem

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	id (^transformer)(id) = ^(id value) {
		if ([value isKindOfClass:[NSString class]]) {
			if ([value length] == 0)
			return (id) nil;
			else
			return [[value componentsSeparatedByString:@""] valueForKey:@"intValue"];
		}
		else
			return (id) @[];
	};
	
	if (!scheme)
		scheme = @{@"messageID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"senderID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"senderName":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"sentDate":@{@"type":@(EVEXMLSchemePropertyTypeDate)},
				   @"title":@{@"type":@(EVEXMLSchemePropertyTypeString)},
				   @"toCorpOrAllianceID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)},
				   @"toCharacterIDs":@{@"type":@(EVEXMLSchemePropertyTypeObject), @"class":[NSArray class], @"transformer":transformer},
				   @"toListID":@{@"type":@(EVEXMLSchemePropertyTypeObject), @"class":[NSArray class], @"transformer":transformer},
				   @"senderTypeID":@{@"type":@(EVEXMLSchemePropertyTypeScalar)}};
	
	return scheme;
}

@end


@implementation EVEMailMessages

+ (NSDictionary*) scheme {
	static NSDictionary* scheme = nil;
	if (!scheme)
		scheme = @{@"messages":@{@"type":@(EVEXMLSchemePropertyTypeRowset), @"class":[EVEMailMessagesItem class]}};
	
	return scheme;
}

@end

File: EVEAPI/EVEAPI/EVEMailMessages.swift
//
//  EVEMailMessages.swift
//  EVEAPI
//
//  Created by Artem Shimanski on 29.11.16.
//  Copyright © 2016 Artem Shimanski. All rights reserved.
//

import UIKit

public class EVEMailMessagesItem: EVEObject {
	public var messageID: Int64 = 0
	public var senderID: Int64 = 0
	public var senderName: String = ""
	public var sentDate: Date = Date.distantPast
	public var title: String = ""
	public var toCorpOrAllianceID: Int64 = 0
	public var toCharacterIDs: [Int64] = []
	public var toListID: [Int64] = []
	public var senderTypeID: Int = 0
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"messageID":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
			"senderID":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
			"senderName":EVESchemeElementType.String(elementName:nil, transformer:nil),
			"sentDate":EVESchemeElementType.Date(elementName:nil, transformer:nil),
			"title":EVESchemeElementType.String(elementName:nil, transformer:nil),
			"toCorpOrAllianceID":EVESchemeElementType.Int64(elementName:nil, transformer:nil),
			"toCharacterIDs":EVESchemeElementType.Int64List(elementName:nil, transformer:nil),
			"toListID":EVESchemeElementType.Int64List(elementName:nil, transformer:nil),
			"senderTypeID":EVESchemeElementType.Int(elementName:nil, transformer:nil),
		]
	}
}


public class EVEMailMessages: EVEResult {
	public var messages: [EVEMailMessagesItem] = []
	
	public required init?(dictionary:[String:Any]) {
		super.init(dictionary: dictionary)
	}
	
	public required init?(coder aDecoder: NSCoder) {
		super.init(coder: aDecoder)
	}
	
	override public func scheme() -> [String:EVESchemeElementType] {
		return [
			"messages":EVESchemeElementType.Rowset(elementName: nil, type: EVEMailMessagesItem.self, transformer: nil)
		]
	}
}
